client/packages/system/src/Item/DetailView/Tabs/ItemVariants/ItemVariantModal.tsx

```diff
@@ -52,9 +52,26 @@ export const ItemVariantModal = ({
           disabled={!isComplete}
           variant="ok"
           onClick={() => {
-            save(draft);
-            success(t('messages.item-variant-saved'))();
-            onClose();
+            save(draft)
+              .then(() => {
+                success(t('messages.item-variant-saved'))();
+                onClose();
+              })
+              .catch(e => {
+                // We create the same error message as we get from the default handler, but prevent duplicates
+                // This avoids the same error message being displayed multiple times, and the only appears once bug...
+                // https://github.com/msupply-foundation/open-msupply/issues/3984
+                if (
```

Comment:
This is a bit of duplication, but kind of a nice hack for the issue.

Also makes me think maybe we want to prevent duplicated on the main API catch, but then allow the errors to occur multiple times. This would mean you won't see too many errors right rather than just waiting an arbitrary length of time?

Reply:
Love the `preventDuplicate` find!

Still get the double error the first time, but that's part of the bigger refactor issue so I think that's fine. Would love to make some time to try the `preventDuplicate` in the common API catch during cooldown period, rather than in each call - then we could just ignore the handled errors here 🔥 

<img width="553" alt="Screenshot 2024-11-04 at 2 22 12 PM" src="https://github.com/user-attachments/assets/2fd668fe-0d9b-49c0-835b-93492db08afb">

Reply:
Oh huh - not even after a page refresh?

Reply:
Hmm you're right I thought it was working on Friday but maybe I hadn't refreshed...